python/pyproject.toml

```diff
@@ -1,3 +1,34 @@
 [build-system]
 requires = ["setuptools"]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "bridgestan"
+readme = "README.md"
+license = { text = "BSD-3-Clause" }
+authors = [
+    { name = "Brian Ward", email = "bward@flatironinstitute.org" },
+    { name = "Bob Carpenter", email = "bcarpenter@flatironinstitute.org" },
+    { name = "Edward Roualdes", email = "eroualdes@csuchico.edu" },
+]
+requires-python = ">=3.9"
+dependencies = ["numpy"]
+dynamic = ["version"]
+
+[tool.setuptools.dynamic]
+version = { attr = "bridgestan.__version.__version__" }
```

Comment:
So this line is `eval()`ed or something like it at install?

Reply:
In a limited way. It can either read a module attribute (like this) or a text file: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata